### PR TITLE
fillkerBulk: init

### PR DESCRIPTION
**Pull request by crispheaney**

### CODE REVIEW

File: src/bots/fillerBulk.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	): {
+		nodesToFill: Array<NodeToFill>;
+		nodesToTrigger: Array<NodeToTrigger>;
+	} {
+		const marketIndex = market.marketIndex;
+
+		const oraclePriceData =
+			this.driftClient.getOracleDataForPerpMarket(marketIndex);
+
+		const fillSlot = this.orderSubscriber.getSlot();
```
Comment (NourAlharithi) [inline src/bots/fillerBulk.ts:31]: might be good to do this.dlobSource.getSlot()

File: src/bots/fillerBulk.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		const nodesToFill = [];
+		const marketIndex = market.marketIndex;
+
+		const oraclePriceData =
+			this.driftClient.getOracleDataForPerpMarket(marketIndex);
+
+		const vAsk = calculateAskPrice(market, oraclePriceData);
+		const vBid = calculateBidPrice(market, oraclePriceData);
+
+		const fillSlot = this.orderSubscriber.getSlot();
```
Comment (NourAlharithi) [inline src/bots/fillerBulk.ts:58]: same thing here w/ dlob.slotSource

File: src/bots/fillerBulk.ts (hunk 1 above)
Comment (crispheaney) [reply src/bots/fillerBulk.ts:32]: added new getMaxSlot since slot subscriber can fall behind

File: src/bots/fillerBulk.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+				if (!takingBidPrice || takingBidPrice.gte(bestAsk)) {
+					nodesToFill.push({
+						node: takingBid,
+						makerNodes: topRestingAsks.filter(
+							(node) => !node.userAccount!.equals(takingBid.userAccount!)
+						),
+					});
+				}
+			}
+		}
```
Comment (wphan) [inline src/bots/fillerBulk.ts:144]: might be insightful to add an `else` here and log whenever there is no bestBid or bestAsk, the user, market index, etc.

**Pull request merged**